libraries/platforms/aws-fpga/bsg_manycore_platform.cpp

```diff
@@ -488,3 +488,38 @@ int hb_mc_platform_fence(hb_mc_manycore_t *mc,
         return HB_MC_SUCCESS;
 }
 
+
+/**
+ * Get the current cycle counter of the Manycore Platform
+ *
+ * @param[in]  mc     A manycore instance initialized with hb_mc_manycore_init()
+ * @param[out] time   The current counter value.
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_platform_get_cycle(hb_mc_manycore_t *mc, uint64_t *time)
```

Comment:
Minor... may be `time` could be a reference to avoid null pointers?

Reply:
Yep. Good idea

Reply:
Added a check in bsg_manycore.cpp (since that is the only place this file gets used)

Reply:
Also added a check in the regression test.